### PR TITLE
added numerous tests for file upload/download

### PR DESCRIPTION
**Pull request by MatakuT**

### CODE REVIEW

**Comment by MatakuT** [comment]
I tried to cut the line length as much as possible to make the code still readable. 
Also, there are two failing tests that will pass as soon as the tickets that I issued will be handled.
The ticket numbers: #2078 , #2075

**Comment by david-yz-liu** [comment]
Cut the line lengths to 80, please.

**Comment by david-yz-liu** [comment]
Comment out the failing tests and add a note to those two issues, then.

**Comment by MatakuT** [comment]
I have one outstanding houndci comment for this issue. But I'm not sure how to fix it.
Avoid using {...} for multi-line blocks.
Block body expression is on the same line as the block start. For the following lines of code:
let(:csv_options) { {
      filename: "#{grade_entry_form.short_identifier}_grades_report.csv",
      disposition: 'attachment',
      type: 'application/vnd.ms-excel'
    } }

File: spec/controllers/grade_entry_forms_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      expect(response.status).to eq(302)
+      expect(flash[:error]).to_not be_empty
+      expect(response).to redirect_to(
+        grades_grade_entry_form_path(grade_entry_form, locale: 'en'))
+    end
+  end
+
+  context 'CSV_Downloads' do
+    let(:csv_data) { grade_entry_form.get_csv_grades_report }
+    let(:csv_options) {
```
Comment (MatakuT) [inline spec/controllers/grade_entry_forms_controller_spec.rb]: I have researched and there seems to be no way to re-write 'let' with 'do end' syntax

**Pull request merged**